### PR TITLE
Add connections section in the configuration file.

### LINKED ISSUE #41: Add `connections` resources.
**Is your feature request related to a problem? Please describe.**

Currently, the `AEA` agent can be run in two ways:
- by interacting directly with the Python library. That is, instantiate an `AEA` instance, set the `mailbox` with the specific `Connection`, and then `start()` the agent.
- by using the `aea` command-line tool (in particular, the `aea run` subcommand), which uses the `OEFConnection`.

**Describe the solution you'd like**

Let the `AEA` agent support multiple connection layers, and be able to run it with one of the connections defined in his resource folder.

More in detail:
- create a new `connections/` folder inside the agent's resource directory.
- support `aea add connection ...` to add a new connection module (and `aea remove` to remove it)
- Let the user be able to specify the connection to use when running `aea run`, using a flexible way to forward the parameters.

**Describe alternatives you've considered**

None.

**Additional context**

None.

### PR DESCRIPTION
**Pull request by marcofavorito**
## Proposed changes

This PR adds a new section to the agent's configuration file, `connections`, to explicitly declare which connections are available to an AEA agent.
Also, the user can switch between connections by using `aea run --connection <connection-name>`.

## Fixes

Fixes #41 

## Types of changes

What types of changes does your code introduce to agents-tac?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [X] I have read the [CONTRIBUTING](../master/CONTRIBUTING.rst) doc
- [X] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments

None.

### CODE REVIEW

File: examples/echo_skill/skill.yaml [hunk 1]
```diff
@@ -3,6 +3,14 @@ authors: Fetch.AI Limited
 version: 0.1.0
 license: Apache 2.0
 url: ""
+connections:
```
[the remark this answers is not in the document]
Comment (marcofavorito) [reply examples/echo_skill/skill.yaml]: Yes! My mistake.

File: aea/channel/oef.py [hunk 2]
```diff
@@ -490,7 +490,7 @@ def disconnect(self) -> None:
             self._connected = False
             self.out_thread.join()
             self.out_thread = None
-            assert self.channel.disconnect(), "Cannot disconnect from OEFChannel."
+            self.channel.disconnect()
```
[the remark this answers is not in the document]
Comment (marcofavorito) [reply aea/channel/oef.py]: the problem is that `self.channel.disconnect()` return `None`, even when the operation succeeded.

File: aea/aea.py [hunk 3]
```diff
@@ -57,12 +58,14 @@ def __init__(self, name: str,
...
         self._directory = directory if directory else str(Path(".").absolute())
 
+        self.mailbox = mailbox
         self._context = Context(self.name, self.outbox)
         self._resources = None  # type: Optional[Resources]
 
     @property
     def context(self) -> Context:
         """Get context."""
+        assert self._context is not None, "Context not initialized."
```
[the remark this answers is not in the document]
Comment (marcofavorito) [reply aea/aea.py]: Yes, you're right. I'll fix that.

File: aea/aea.py [hunk 4]
```diff
@@ -34,6 +34,7 @@ class AEA(Agent):
     """This class implements an autonomous economic agent."""
 
     def __init__(self, name: str,
+                 mailbox: MailBox,
```
[the remark this answers is not in the document]
Comment (marcofavorito) [reply aea/aea.py]: :+1:

File: aea/channel/oef.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (DavidMinarsch) [reply aea/channel/oef.py]: Yeah ok. Annoying but let's leave that then.

**Pull request merged**